src/test/java/com/opencart/tests/currency/ChangeCurrencyTest.java

```diff
@@ -0,0 +1,52 @@
+package com.opencart.tests.currency;
+
+import com.opencart.driver.DriverRepository;
+import com.opencart.enums.CurrencyName;
+import com.opencart.navigation.Navigation;
+import com.opencart.steps.AdminDashboardBL;
+import com.opencart.steps.AdminLoginPageBL;
+import com.opencart.steps.MainPageBL;
+import com.opencart.tests.BaseTest;
+import org.testng.annotations.*;
+
+import static com.opencart.enums.URLs.ADMIN_BASE_URL;
+import static com.opencart.enums.URLs.BASE_URL;
+
+/**
+ * Test checks if the cost in EUR, USD, GBP, UAH of all products on the website's main page is correct"
+ */
+public class ChangeCurrencyTest extends BaseTest {
+
+    @BeforeClass
+    public void getCurrenciesValuesFromAdminPage() {
+        new Navigation().navigateToURrl(ADMIN_BASE_URL.getValue());
+        new AdminLoginPageBL().adminLogin();
+        new AdminDashboardBL()
+                .clickOnSidebarButton("System")
+                .clickOnSidebarButton("Localisation")
+                .clickOnSidebarButton("Currencies")
+                .getCurrencyValues ();
```

Comment:
According to changes in that method which I describe above, you will get the following: 
Map<String, String> currencies = adminDashBoard.getCurrencyValues ();

And, also, you do not need to use a data-provider here, because you are hardcoding currencies there. It is not right, because you can not be sure that exactly the same values as into the data-provided will be returned by getCurrencyValues () method.

To sum up:
1. Remove data-provider
2. Update verifyAllProductsPrices() method, it should take Map<String, String> currency values got from admin page.

Reply:
You have not finished make changes according to this comment yet

Reply:
I suggest living dataProvider for USD, EUR, GBP currencies because they are standard for opencart and already implemented. Or it is better to write separate tests for USD, EUR, GBP currencies?